packages/core/src/rules/common/no-required-schema-properties-undefined.ts

```diff
@@ -0,0 +1,30 @@
+import { Oas3Rule } from '../../visitors';
+import { Oas3Schema, Oas3_1Schema } from '../../typings/openapi';
+import { Oas2Schema } from 'core/src/typings/swagger';
+import { UserContext } from 'core/src/walk';
+
+export const NoRequiredSchemaPropertiesUndefined: Oas3Rule = () => {
+  return {
+    Schema: {
+      enter(schema: Oas3Schema | Oas3_1Schema | Oas2Schema, { location, report }: UserContext) {
+        if (schema?.required) {
+          const missingRequiredProperties: Array<string> = schema.required.filter((property) => {
+            return !schema.properties || !Object.keys(schema.properties!).includes(property);
+          });
+
+          if (missingRequiredProperties.length) {
+            const reportMessage =
+              missingRequiredProperties.length > 1
+                ? `Required properties are undefined: ${missingRequiredProperties.join(', ')}.`
+                : `Required property ${missingRequiredProperties.join()} is undefined.`;
+
+            report({
+              message: reportMessage,
+              location: location.key(),
```

Comment:
It would be better to report on each required item that doesn't have a correspondent property. You can go deeper using the `child` syntax like this: 
```ts
location.child('reuqired')...
```